--- rust/extensions.bzl
+++ rust/extensions.bzl
@@ -63,8 +63,12 @@ def _rust_impl(module_ctx):
         toolchain = toolchains[0]
         if len(toolchain.versions) == 1:
             version = toolchain.versions[0]
+            iso_date = ""
+            if version.startswith(("beta", "nightly")):
+                version, _, iso_date = version.partition("/")
         elif not toolchain.versions:
             version = None
+            iso_date = ""
         else:
             fail(_HOST_TOOL_ERR % "multiple toolchain versions are provided")
         host_tools = struct(
@@ -75,6 +79,7 @@ def _rust_impl(module_ctx):
             sha256s = toolchain.sha256s,
             urls = toolchain.urls,
             version = version,
+            iso_date = iso_date,
         )
     else:
         fail("Multiple host_tools were defined in your root MODULE.bazel")
@@ -91,6 +96,7 @@ def _rust_impl(module_ctx):
         sha256s = host_tools.sha256s,
         urls = host_tools.urls,
         version = host_tools.version or rust_common.default_version,
+        iso_date = host_tools.iso_date,
     )
 
     for toolchain in toolchains:
